contracts/HtsSystemContract.sol

```diff
@@ -940,4 +954,14 @@ contract HtsSystemContract is IHederaTokenService {
         assembly { sstore(slot, approved) }
         emit IERC721.ApprovalForAll(sender, operator, approved);
     }
+
+    function _keyExists(uint keyType, TokenInfo memory tokenInfo) private pure returns (bool) {
```

Comment:
let's move this into its own `library` so we can put all key related stuff in there.

name: `KeyLib`? `KeyMgmt`? other alternatives?